src/playlist-controller.js

```diff
@@ -977,23 +977,9 @@ export class PlaylistController extends videojs.EventTarget {
     // Don't need to reset audio as it is reset when media changes.
     // We resetLoaderProperties separately here as we want to fetch init segments if
     // necessary and ensure we're not in an ended state when we switch playlists.
-    this.resetMainLoaderReplaceSegments();
-  }
-
-  /**
-   * Sets the replaceUntil flag on the main segment soader to the buffered end
-   * and resets the main segment loaders properties.
-   */
-  resetMainLoaderReplaceSegments() {
-    const buffered = this.tech_.buffered();
-    const bufferedEnd = buffered.length ? buffered.end(buffered.length - 1) : 0;
-
-    // Set the replace segments flag to the buffered end, this forces fetchAtBuffer
-    // on the main loader to remain, false after the resetLoader call, until we have
-    // replaced all content buffered ahead of the currentTime.
-    this.mainSegmentLoader_.replaceSegmentsUntil = bufferedEnd;
-    this.mainSegmentLoader_.resetLoaderProperties();
-    this.mainSegmentLoader_.resetLoader();
+    this.delegateLoaders_('all', ['pause']);
+    this.resetAllSegmentLoaders_();
+    this.load();
```

Comment:
should we use `delegateLoaders('all', ['resetEverything'])` and `delegateLoaders('all', ['load'])` ?

Reply:
I actually tried something similar and couldn't get it to work the first time. Will try again before merging this.